### PR TITLE
Forms

### PR DESCRIPTION
**Pull request by AKeeganDev**
## This pull request adds the following features:

- Users can add comments to posts
- Users can now like posts
- logic has been added to various actions for communicating to the view and allowing the creation of posts and comments
- form partials created to avoid repeating code

## To setup database and use the site

Please run the following commands in the project folder after cloning the repo:

`bundle install`
`rails db:create`
`rails db:migrate`

## Access the rails console with `rails c` and run the following commands to add data to the database
`new_user = User.create(name: 'Test User', photo: 'www.google.com', bio: 'Student from Test Country')`

You will now have 1 user which is all that is required to use the features of this pull request

To host the site on your computer and to see it in your browser run `rails s` and navigate to http://localhost:3000/ in your browser

### CODE REVIEW

File: app/views/posts/show.html.erb [hunk 1]
```diff
@@ -1,8 +1,13 @@
...
   <div class="post_comments">
-    <%for a in 1..5 do%>
-      <div><%="Username: Comment #{a}"%></div>
+  <div>Comments </div>
+    <%@comments.each do |comment|%>
+      <div><%="#{comment.user.name}: #{comment.text}"%></div>
     <%end%>
   </div>
+<% end %>
+<button type='button'><%= link_to "Create new post", new_user_post_comment_path(@user, @post) %></button>
```
Comment (rustam-tolipov) [inline app/views/posts/show.html.erb]: - [x] Please change create post link to create comment.

File: config/routes.rb [hunk 2]
```diff
@@ -6,8 +6,18 @@
...
+    end
+  end
+
+  # get '/users/:user_id/posts' => 'posts#index', as: :user_posts
+  # post 'posts/create' => 'posts#create', as: :create_user_post
+  # get '/users/:user_id/posts/new' => 'posts#new', as: :new_user_post
+  # get '/users/:user_id/posts/:id' => 'posts#show', as: :user_post
+  # get '/users' => 'users#index', as: :users
+  # get '/users/:id' => 'users#show', as: :user
+  # post 'likes/create' => 'likes#create', as: :create_user_post
```
Comment (rustam-tolipov) [inline config/routes.rb]: - [x] Kindly remove commented codes from your file.

File: app/controllers/application_controller.rb [hunk 3]
```diff
@@ -1,5 +1,5 @@
 class ApplicationController < ActionController::Base
   def current_user
-    @user = User.first
+    User.first
```
Comment (rustam-tolipov) [inline app/controllers/application_controller.rb:3]: - [ ] [OPTIONAL] I like the description you wrote for this PR. It's really well written. Personal suggestion, have you used `rails db:seed` before? If not, here is the chance to test it. You can look through about it in [here](https://blog.devgenius.io/how-to-seed-with-ruby-on-rails-%EF%B8%8F-1d2dceda3e7d#:~:text=%23%20db/seed.rb,puts%20%22Seeding%20done.%22).

**Review by rustam-tolipov — CHANGES_REQUESTED** [review]
Hi @AKeeganDev, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
_Your project is almost complete!_
_No Linters errors_


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/application_controller.rb (hunk 3 above)
Comment (AKeeganDev) [reply app/controllers/application_controller.rb:3]: @Rustamxon7 I have not seen this before thanks for sharing! I appreciate the extra knowledge :)

**Review by elisha2kyakpo1 — APPROVED** [review]
Hi 👋 @AKeeganDev, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- You have done a great job working on the requested changes✔️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**